### ISSUE TITLE
NUM_WARMUP_UPDATES multiplied by number of GPU's during multi-GPU training

### ISSUE THREAD

**Issue by abhisirka2001** [SWivid/F5-TTS#1259]
### Checks

- [x] This template is only for usage issues encountered.
- [x] I have thoroughly reviewed the project documentation but couldn't find information to solve my problem.
- [x] I have searched for existing issues, including closed ones, and couldn't find a solution.
- [x] I am using English to submit this issue to facilitate community communication.

### Environment Details

Hi,

I’m finetuning F5 on ~47k hours of multilingual data using 8 H100s. I set:

Base ckpt: https://huggingface.co/SWivid/F5-TTS/blob/main/F5TTS_Base/model_1200000.pt
`NUM_WARMUP_UPDATES=20000`

but during training the effective warmup appears to be 20k × 8 = 160k steps.

Is warmup counted per-GPU or globally? Is this scaling with world size expected, or should warmup be specified as global steps?

Also, do these params look reasonable for this scale?

```
BATCH_SIZE_PER_GPU=35000
LR=7e-5
EPOCHS=10
```

Any guidance on warmup/LR for large multi-GPU runs would be appreciated. Thanks!

### Steps to Reproduce

No response

### ✔️ Expected Behavior

No response

### ❌ Actual Behavior

No response

**Comment by SWivid** [comment]
> but during training the effective warmup appears to be 20k × 8 = 160k steps.

i think it's always the number you set in the .yaml config file
https://github.com/SWivid/F5-TTS/blob/c279a2b7d573bc87a4eac6f6c08882188dea1b10/src/f5_tts/model/trainer.py#L313-L318

> Any guidance on warmup/LR for large multi-GPU runs would be appreciated. Thanks!

just use the default setting if you are not sure, change epochs up to your need

**Comment by abhisirka2001** [comment]
Based on the code, num_warmup_updates is multiplied by num_processes (8 in my case), so I should explicitly set it to the value I want for global warmup updates.

should i use?

`warmup_updates = 20k (global, as in the paper)` ?

LR as per paper

Planning to train for ~2M updates

Thanks for the clarification.

reference to why i am saying it is 160K in my case:

<img width="628" height="78" alt="Image" src="https://github.com/user-attachments/assets/52bdfef0-3e2c-4f2b-bbf4-940c2cb35e3c" />

**Comment by abhisirka2001** [comment]
Got it
The plot of lr vs steps is accurately reaching peak at 20K steps.
The warmup updates specified in the config is the true warmup updates.
